composer.json

```diff
@@ -27,7 +27,8 @@
     "require-dev": {
         "phpunit/phpunit": "^8.4",
         "psr/http-factory": "^1.0",
-        "nyholm/psr7": "^1.0"
+        "nyholm/psr7": "^1.0",
+        "yiisoft/router-fastroute": "^3.0@dev"
```

Comment:
Is there a way to avoid using concrete implementation for testing? That doesn't look good at all.

Reply:
> Is there a way to avoid using concrete implementation for testing? That doesn't look good at all.

If we remove the adding of group middlewares from the FastRoute class, this package will not be needed.